src/entitysdk/common.py

```diff
@@ -1,12 +1,42 @@
 """Common stuff."""
 
-from dataclasses import dataclass
+import re
+from typing import Self
 from uuid import UUID
 
+from pydantic import BaseModel
 
-@dataclass
-class ProjectContext:
+from entitysdk.exception import EntitySDKError
+from entitysdk.typedef import DeploymentEnvironment
+
+VLAB_URL_PATTERN = re.compile(
+    r"^https:\/\/(?:(?P<staging>staging)\.)?openbraininstitute\.org"
+    r"\/app\/virtual-lab\/lab\/(?P<vlab>[0-9a-f]{8}(?:-[0-9a-f]{4}){3}-[0-9a-f]{12})"
+    r"\/project\/(?P<proj>[0-9a-f]{8}(?:-[0-9a-f]{4}){3}-[0-9a-f]{12})(?:\/.*)?$"
```

Comment:
Wouldn't the regexp be slightly more readable if we move out and reuse the UUID regexp?